interface/patient_file/encounter/find_code_popup.php

```diff
@@ -49,200 +49,169 @@
-    <?php } ?>
-
-    </center>
+        <?php if (!empty($allowed_codes)) { ?>
+        <form class="form-inline" method='post' name='theform' action='find_code_popup.php<?php echo $string_target_element ?>codetype=<?php echo attr_url($codetype) ?>'>
+        <?php } else { ?>
+        <form class="form-inline"  method='post' name='theform' action='find_code_popup.php<?php echo $string_target_element ?>'>
+        <?php } ?>
+            <input type="hidden" name="csrf_token_form" value="<?php echo attr(CsrfUtils::collectCsrfToken()); ?>" />
+            <div class="form-group">
+                <?php
+                if (!empty($allowed_codes)) {
+                    if (count($allowed_codes) === 1) {
+                        echo "<input class='form-control' type='text' name='form_code_type' value='" . attr($codetype) . "' readonly />\n";
+                    } else { ?>
+                    <select class='form-control' name='form_code_type'>
+                        <?php
+                        foreach ($allowed_codes as $code) {
+                            if (empty($code_types[$code]['label'])) {
+                                continue;
+                            }
+                            $selected_attr = ($default == $code) ? " selected='selected'" : '';
+                            ?>
+                        <option value='<?php echo attr($code) ?>'<?php echo $selected_attr ?>><?php echo xlt($code_types[$code]['label']) ?></option>
+                        <?php } ?>
+                    </select>
+                    <?php }
+                } else {
+                    // No allowed types were specified, so show all.
+                    echo "<select class='form-control' name='form_code_type'";
+                    echo ">\n";
+                    foreach ($code_types as $key => $value) {
+                        if (empty($value['label'])) {
+                            continue;
+                        }
+                        echo "<option value='" . attr($key) . "'";
+                        if ($default == $key) {
+                            echo " selected";
+                        }
+                        echo ">" . xlt($value['label']) . "</option>\n";
+                    }
+                    echo "<option value='PROD'";
+                    if ($default == 'PROD') {
+                        echo " selected";
+                    }
+                    echo ">" . xlt("Product") . "</option>\n";
+                    echo "</select>\n";
+                }
+                ?>
+                <div class="input-group mt-1">
+                    <input type='text' class='form-control' name='search_term' id="searchTerm"
+                        value='<?php echo attr($_REQUEST['search_term']); ?>'
+                        title='<?php echo xla('Any part of the desired code or its description'); ?>'
+                        placeholder="<?php echo xlt('Search for'); ?>" />
```

Comment:
xla
(finally, i get to point out a escaping thing :) )

Reply:
you know me, I rarely pay a lot of attention to that...